File: Python/Programmers/Level 2/다음 큰 숫자.py
# 문제에 나온 조건 그대로 따라가면 됨

def solution(n):
    # 조건 2에서 n보다 다음 큰 수와 n을 2진수로 변환했을 때 1의 개수가 같아야 하므로 n을 2진수로 변환 후 1의 개수를 저장
    check = bin(n).count("1")
    while True:
        n += 1 # 조건 1
        if check == bin(n).count("1"): # 조건1, 조건2를 만족하는 수 중 제일 작은 수 반환
            return n

File: JavaScript/Programmers/Level 2/다음 큰 숫자.js
// n이 1씩 커질때마다 n을 이진수로 바꿨을 때 1의 개수가 같은지 확인

// function solution(n) {
//     // toString(2) 메소드를 이용하여 n을 2진수로 변환 후 나머지 문법을 이용하여 배열로 변환
//     // 이후 filter 메소드를 이용하여 1의 개수를 카운팅
//     const init = [...n.toString(2)].filter((bin) => bin === "1").length;
//     while (true) {
//         // n이 1씩 커질때마다 2진수로 변환 후 1의 개수 카운팅
//         const next = [...(++n).toString(2)].filter((bin) => bin === "1").length;
//         if (init === next) return n; // 1의 개수가 같으면 n return
//     }
// }

// function solution(n) {
//     const init = n.toString(2).match(/1/g).length; // 정규식을 사용하여 1의 개수를 구함, match 메소드는 pattern과 일치하는 항목들을 배열로 변환
//     while (true) {
//         if (init === (++n).toString(2).match(/1/g).length) return n; // 1의 개수가 같으면 n return
//     }
// }

function solution(n) {
    const init = n.toString(2).match(/1/g).length;
    while (init !== (++n).toString(2).match(/1/g).length) {} // while 조건식에 1의 개수 비교하는 조건식을 넣고 body를 비워 반복
    return n; // 1의 개수가 같으면 while문 탈출 후 n return
}
